pykern/util.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+"""utils for pykern
+
+:copyright: Copyright (c) 2023 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+from pykern import pkconfig
+from pykern import pkinspect
+from pykern import pkio
+from pykern.pkdebug import pkdc, pkdexc, pkdlog, pkdp
+import os.path
+import sys
+
+
+_DEFAULT_ROOT = "run"
+
+
+def cfg_db_dir(db_root_env_name):
+    """Generate pkconfig parser function with errors for db's env var name
+
+    Args:
+        db_root_env_name (str): env var name for run dir root
+
+    Returns:
+        function: parser function for pkconfig.init
+    """
+    _NOT_ABS_ERROR = "{}: " + f"{db_root_env_name} must be absolute"
+    _NO_DIR_ERROR = "{}: " + f"{db_root_env_name} must be a directory and exist"
+
+    def cfg_db_dir_parser(v):
+        """Config value or root package's parent or cwd with `_DEFAULT_ROOT`"""
+        if not os.path.isabs(v):
+            pkconfig.raise_error(_NOT_ABS_ERROR.format(v))
+        if not os.path.isdir(v):
+            pkconfig.raise_error(_NO_DIR_ERROR.format(v))
+        return pkio.py_path(v)
+
+    return cfg_db_dir_parser
+
+
+def init_db_dir(package_object):
+    """Initialize run dir
+
+    Args:
+        package_object (object): python object who's root_package will hold the run dir
+
+    Returns:
+        py.path.local: absolute path to run dir
+    """
+    r = (
```

Comment:
Need to assert dev mode. This should be called something like `dev_run_dir`. It's more than "db" in that it is the root of where all run artifacts are held.